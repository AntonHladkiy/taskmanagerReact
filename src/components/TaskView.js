import React  from 'react';
import {Link , Redirect} from "react-router-dom";


const TaskView = props => {
    const task=props.currentTask;
    if(props.currentTask.description===""){
        return <Redirect to="/"/>
    }
    return (
            <div>
            <h3>{task.title}</h3>
            <dl className="row">
                <dt className="col-sm-3">Description:</dt>
                <dd className="col-sm-9">{task.description}</dd>
                <dt className="col-sm-3">Priority:</dt>
                <dd className="col-sm-9">{task.priority}</dd>
                <dt className="col-sm-3">Due date:</dt>
                <dd className="col-sm-9">{task.dueDate}</dd>
                <dt className="col-sm-3">Completed:</dt>
                { task.done &&
                <dd className="col-sm-9"><input type="checkbox" checked disabled/></dd> }
                { ! task.done &&
                <dd className="col-sm-9"><input type="checkbox" disabled/></dd>
                }
            </dl>
            <Link to="/"><button className={"btn btn-danger mt-2 w-25"}>Back</button></Link>
            </div>
    )

};
export default TaskView;