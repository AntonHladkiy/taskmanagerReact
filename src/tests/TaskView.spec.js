import React from 'react';
import { shallow } from 'enzyme';
import TaskView from '../components/TaskView';
const completedTask = {
        title:'title',
        description:'desc',
        dueDate:'',
        priority:1,
        done:true
};
const notCompletedTask = {
    title:'title',
    description:'desc',
    dueDate:'',
    priority:1,
    done:false
};
const emptyTask = {
    title:'',
    description:'',
    dueDate:'',
    priority:'',
    done:''
};
describe('TaskView with completed task', () => {
    let wrapped = shallow(<TaskView currentTask={completedTask}></TaskView>);
    it('should render the TaskView Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
    it('renders the TaskView children', () => {
        expect(wrapped.find('h3').text()).toEqual(completedTask.title);
    });
});
describe('TaskView  with not completed task', () => {
    let wrapped = shallow(<TaskView currentTask={notCompletedTask}></TaskView>);
    it('should render the TaskView Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
    it('renders the TaskView children', () => {
        expect(wrapped.find('h3').text()).toEqual(notCompletedTask.title);
    });
});
describe('TaskView with empty task', () => {
    let wrapped = shallow(<TaskView currentTask={emptyTask}></TaskView>);
    it('should render the TaskView Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});