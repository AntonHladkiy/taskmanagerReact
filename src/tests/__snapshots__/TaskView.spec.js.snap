// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView should render the TaskView Component correctly 1`] = `
<div>
  <h3>
    title
  </h3>
  <dl
    className="row"
  >
    <dt
      className="col-sm-3"
    >
      Description:
    </dt>
    <dd
      className="col-sm-9"
    >
      desc
    </dd>
    <dt
      className="col-sm-3"
    >
      Priority:
    </dt>
    <dd
      className="col-sm-9"
    >
      1
    </dd>
    <dt
      className="col-sm-3"
    >
      Due date:
    </dt>
    <dd
      className="col-sm-9"
    />
    <dt
      className="col-sm-3"
    >
      Completed:
    </dt>
    <dd
      className="col-sm-9"
    >
      <input
        checked={true}
        disabled={true}
        type="checkbox"
      />
    </dd>
  </dl>
  <Link
    to="/"
  >
    <button
      className="btn btn-danger mt-2 w-25"
    >
      Back
    </button>
  </Link>
</div>
`;
